v2/tools/generator/internal/codegen/pipeline/create_arm_types.go

```diff
@@ -418,6 +418,8 @@ func (c *armTypeCreator) createARMProperty(
 		// See https://azure.github.io/azure-service-operator/design/adr-2023-04-patch-collections/ for how we're solving it.
 		result = result.WithoutTag("json", "omitempty")
 
+	case config.ExplicitEmptyCollections:
+		fallthrough
 	case config.ExplicitCollections:
 		// With PayloadType 'explicitCollections' we remove the `omitempty` tag from arrays and maps, because we always
```

Comment:
```suggestion
		// With PayloadType 'explicitCollections' or 'explicitEmptyCollections' we remove the `omitempty` tag from arrays and maps, because we always
```

Reply:
Done